src/TestInfrastructure/Helpers/MefTestHelpers.cs

```diff
@@ -180,5 +180,27 @@ public static void Compose(Assembly exportingAssembly, params object[] importers
             using var container = new CompositionContainer(catalog);
             container.Compose(batch);
         }
+
+        /// <summary>
+        /// Check if type has creation policy non shared.
+        /// </summary>
+        public static bool IsMefComponentNonShared(Type type)
+        {
+            var customAttributes = type.GetCustomAttributes(typeof(PartCreationPolicyAttribute), true);
+            var partCreationPolicyAttribute = (PartCreationPolicyAttribute)customAttributes[0];
```

Comment:
This will fail with an array out-of-bounds exception if it is called on a type that doesn't have the attribute. Test failures might be easier to debug with a more explicit assertion e.g. 
`customAttributes.Should().HaveCount(1)`.